### PR TITLE
Added improvements to sleep feature

### PR DESCRIPTION
**Pull request by DaDevChia**
Improved code quality, tests and general readability for all classes in sleep.

### CODE REVIEW

**Review by skylee03 — COMMENTED** [review]
LGTM!
Suggestion: Rename the test methods according to our [coding standard](https://se-education.org/guides/conventions/java/basic.html).

File: src/main/java/athleticli/commands/sleep/ListSleepCommand.java [hunk 1]
```diff
@@ -2,15 +2,25 @@
...
 import athleticli.data.sleep.SleepList;
+import athleticli.ui.Message;
 
 public class ListSleepCommand extends Command {
 
+    /**
+     * Lists all the sleep records in the sleep list.
+     * @param data The current data containing the sleep list.
+     * @return The message which will be shown to the user.
+     */
```
Comment (nihalzp) [inline src/main/java/athleticli/commands/sleep/ListSleepCommand.java:14]:
```suggestion
    /**
     * Lists all the sleep records in the sleep list.
     *
     * @param data The current data containing the sleep list.
     * @return The message which will be shown to the user.
     */
```

File: src/main/java/athleticli/data/sleep/Sleep.java [hunk 2]
```diff
@@ -1,15 +1,33 @@
...
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("dd-MM-YYYY HH:mm");
+
+    private LocalDateTime from;
+    private LocalDateTime to;
+
+    /**
+     * Constructor for Sleep.
+     * @param from Start time of the sleep.
+     * @param to End time of the sleep.
+     */
```
Comment (nihalzp) [inline src/main/java/athleticli/data/sleep/Sleep.java:20]:
```suggestion
    /**
     * Constructor for Sleep.
     *
     * @param from Start time of the sleep.
     * @param to End time of the sleep.
     */
```

File: src/main/java/athleticli/data/sleep/Sleep.java [hunk 3]
```diff
@@ -1,15 +1,33 @@
...
+    public Sleep(LocalDateTime from, LocalDateTime to) {
         this.from = from;
         this.to = to;
     }
 
+    /**
+     * toString method for Sleep.
+     * @return String representation of the sleep record.
+     */
```
Comment (nihalzp) [inline src/main/java/athleticli/data/sleep/Sleep.java:29]:
```suggestion
    /**
     * toString method for Sleep.
     *
     * @return String representation of the sleep record.
     */
```

File: src/main/java/athleticli/data/sleep/SleepList.java [hunk 4]
```diff
@@ -2,7 +2,14 @@
...
 import java.util.ArrayList;
 
+/**
+ * Represents a list of sleep records.
+ */
 public class SleepList extends ArrayList<Sleep> {
+    /**
+     * toString method for SleepList.
+     * @return String representation of the sleep list.
+     */
```
Comment (nihalzp) [inline src/main/java/athleticli/data/sleep/SleepList.java:12]:
```suggestion
    /**
     * toString method for SleepList.
     *
     * @return String representation of the sleep list.
     */
```

File: src/main/java/athleticli/commands/sleep/EditSleepCommand.java [hunk 5]
```diff
@@ -1,31 +1,49 @@
...
+    /**
+     * Edits the sleep record at the specified index.
+     * @param data The current data containing the sleep list.
+     * @return The message which will be shown to the user.
+     */
     public String[] execute(Data data) {
         SleepList sleepList = data.getSleeps();
         Sleep oldSleep = sleepList.get(index - 1);
         Sleep newSleep = new Sleep(from, to);
         sleepList.set(index - 1, newSleep);
```
[the remark this answers is not in the document]
Comment (DaDevChia) [reply src/main/java/athleticli/commands/sleep/EditSleepCommand.java]: Thanks for the bug catch, will definitely add that as well.

File: src/main/java/athleticli/commands/sleep/DeleteSleepCommand.java [hunk 6]
```diff
@@ -2,35 +2,47 @@
...
+     * Deletes the sleep record at the specified index.
+     * @param data The current data containing the sleep list.
+     * @return The message which will be shown to the user.
+     */
     public String[] execute(Data data) {
         SleepList sleepList = data.getSleeps();
         if (index > sleepList.size() || index < 1) {
             return new String[] {
-                "Invalid index. Please enter a valid index."
+                Message.MESSAGE_SLEEP_DELETE_INVALID_INDEX
```
[the remark this answers is not in the document]
Comment (DaDevChia) [reply src/main/java/athleticli/commands/sleep/DeleteSleepCommand.java]: Yes agreed will refactor this code to throw the corresponding exception

**Review by DaDevChia — COMMENTED** [review]
Implemented changes for code review and added exception handling for OOBE in Sleep Edit and Delete commands

**Pull request merged**